api-put-object.go

```diff
@@ -92,12 +155,19 @@ func (c Client) PutObject(bucketName, objectName string, data io.Reader, size in
 		return c.putAnonymous(bucketName, objectName, data, size, contentType)
 	}
 
+	// putSmall object.
+	if size < minimumPartSize || size > 0 {
```

Comment:
I was thinking of a different approach. There are three types of reader.
1) io.Reader
2) *File
3) minio.Object
With this approach, we can get rid of caching tmpfiles.